contribute/design/1.x/upgrade/volume-pools-upgrade.md

```diff
@@ -0,0 +1,499 @@
+    details via a config map. 
+
+  While the above approach was good enough to be automated via `kubectl`, 
+  the steps that can be executed as part of upgrade were limited
+  by the features or constructs available within the CAST/RunTasks. 
+
+  Another short-coming of the CAST/RunTasks is the lack of developer
+  friendly constructs for rapidly writing new CAST/RunTasks. This was
+  severely impacting the time taken to push out new releases. 
+
+### High Level Design
+
+#### Upgrade Job Example
+
+Here is an example of Kubernetes Job spec for upgrading the jiva volume.
+```
+#This is an example YAML for upgrading jiva volume. 
+#Some of the values below needs to be changed to
+#match your openebs installation. The fields are
+#indicated with VERIFY
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  #VERIFY that you have provided a unique name for this upgrade job.
+  #The name can be any valid K8s string for name. This example uses
+  #the following convention: jiva-vol-<flattened-from-to-versions>-<pv-name>
+  name: jiva-vol-100110-pvc-713e3bb6-afd2-11e9-8e79-42010a800065
+  #VERIFY the value of namespace is same as the namespace where openebs components
+  # are installed. You can verify using the command:
+  # `kubectl get pods -n <openebs-namespace> -l openebs.io/component-name=maya-apiserver`
+  # The above command should return status of the openebs-apiserver.
+  namespace: openebs
+spec:
+  backoffLimit: 4
+  template:
+    spec:
+      #VERIFY the value of serviceAccountName is pointing to service account
+      # created within openebs namespace. Use the non-default account.
+      # by running `kubectl get sa -n <openebs-namespace>`
+      serviceAccountName: openebs-maya-operator
+      containers:
+      - name:  upgrade
+        args: 
+        - "jiva-volume"
+        - "--from-version=1.0.0"
+        - "--to-version=1.1.0-RC2"
+        #VERIFY that you have provided the correct cStor PV Name
+        - "--pv-name=pvc-713e3bb6-afd2-11e9-8e79-42010a800065"
+        #Following are optional parameters
+        #Log Level
+        - "--v=4"
+        #DO NOT CHANGE BELOW PARAMETERS
+        env:
+        - name: OPENEBS_NAMESPACE
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.namespace
+        tty: true 
+        image: openebs/m-upgrade:dev-1000110RC2-072703
+      restartPolicy: Never
+---
+```
+
+Execute the Upgrade Job Spec
+```
+$ kubectl apply -f jiva-vol-100110-pvc713.yaml
+```
+
+You can check the status of the Job using commands like:
+```
+$ kubectl get job -n openebs
+$ kubectl get pods -n openebs #to check on the name for the job pod
+$ kubectl logs -n openebs jiva-upg-100111-pvc-713e3bb6-afd2-11e9-8e79-42010a800065-bgrhx
+```
+
+#### UpgradeTask CR Example
+
+Here is an example of UpgradeTask for upgrading Jiva Volume. The 
+conditions are added by the Upgrade Job
+
+```
+apiVersion: openebs.io/v1alpha1
+kind: UpgradeTask
```

Comment:
The kind is confusing to me based on my understanding of this OEP.
This custom resource reflects a status or result. However, the kind indicates something will be done.

Reply:
updated the sections to indicate it is both input and result.